src/main/java/org/primefaces/util/CalendarUtils.java

```diff
@@ -530,7 +530,11 @@ public static LocalDateTime toLocalDateTime(ZoneId zoneId, String isoDateString)
      */
     public static Temporal now(UICalendar uicalendar) {
         boolean hasTime = uicalendar.hasTime();
+        boolean timeOnly = uicalendar.isTimeOnly();
         ZoneId zone = calculateZoneId(uicalendar.getTimeZone());
+        if (hasTime && timeOnly) {
```

Comment:
One last thing. :-)
hasTime = date and time
timeOnly = time
(I know this is a bit confusing.)
--> `if (timeOnly) {`